terraform/projects/app-ecs-instances/instance-user-data.tpl

```diff
@@ -1,23 +1,45 @@
 #!/bin/bash
 # Attach EBS volume to instance
 echo "[$(date '+%H:%M:%S %d-%m-%Y')] installing dependencies for volume attaching"
-sudo yum install -y aws-cli wget
+sudo yum install -y aws-cli wget jq
 
 REGION="${region}"
 DEVICE="xvdf"
-VOLUME_IDS="${volume_ids}"
 
 echo "[$(date '+%H:%M:%S %d-%m-%Y')] finding current instance ID"
-INSTANCE_ID="`wget -q -O - http://169.254.169.254/latest/meta-data/instance-id`"
+INSTANCE_ID=$(wget -q -O - http://169.254.169.254/latest/meta-data/instance-id)
 
 echo "[$(date '+%H:%M:%S %d-%m-%Y')] finding volume to attach"
 AZ="$(wget -q -O - http://169.254.169.254/latest/meta-data/placement/availability-zone)"
-VOLUME_ID="$(aws ec2 describe-volumes --filters Name=availability-zone,Values=$AZ --volume-ids $VOLUME_IDS --region $REGION --query Volumes[*].VolumeId --output text)"
+VOLUME_ID="$(aws ec2 describe-volumes --filters Name=availability-zone,Values="$AZ" --volume-ids ${volume_ids} --region "$REGION" --query Volumes[*].VolumeId --output text)"
 
-echo "[$(date '+%H:%M:%S %d-%m-%Y')] attaching volume: $VOLUME_ID"
-aws ec2 attach-volume --volume-id $VOLUME_ID --instance-id $INSTANCE_ID --device /dev/$DEVICE --region $REGION
+count=0
+DISK_AVAILABILITY="NA"
+until [ "$DISK_AVAILABILITY" = available ]; do
+    if [[ $count -le 10 ]]
+    then
+        sleep 10;
+        echo "Sleeping for some time"
+        count=$((count+1));
+        echo "In while loop"
+        DISK_AVAILABILITY=$(aws ec2 describe-volumes --region "$REGION" --filters Name=volume-id,Values="$VOLUME_ID" --query Volumes[0].State --output text)
+    else
+        break
+    fi
+done
+
+case $DISK_AVAILABILITY in
+        available)
+            echo "[$(date '+%H:%M:%S %d-%m-%Y')] attaching volume: $VOLUME_ID"
+            aws ec2 attach-volume --volume-id "$VOLUME_ID" --instance-id "$INSTANCE_ID" --device /dev/"$DEVICE" --region "$REGION";
+         ;;
+        *)
+            shutdown -h now;
+         ;;
+esac
 
 # Waiting for volume to finish attaching
+#I do not know why this does not work on the instance
```

Comment:
Is this working now?

Reply:
yeah I've taken this line out about three times, kept coming up after rebasing, is gone for good now (hopefully)